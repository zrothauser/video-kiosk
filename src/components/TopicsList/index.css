.b-topics-list {
  text-transform: uppercase;
  letter-spacing: 0.1em;
  font-size: 1.5625vw;
  line-height: 2.2;
}

.b-topics-list__title {
  font-weight: 700;
  font-size: 0.8333333333vw;
  margin: 0;
  box-sizing: border-box;
  padding: 2.4479166667vw 0 0 0;
  height: 5.2083333333vw;
  letter-spacing: .16em;
}

.b-topics-list__list {
  margin: 0;
  padding: 0;
  list-style: none;
}

.b-topics-list__item {
  margin: 0;
  padding: 0;
}

.b-topics-list__link {
  color: var(--color__grey--darker);
  opacity: 1;
}

.b-topics-list__link--selected {
  color: var(--color__white);
  font-weight: normal;
}
