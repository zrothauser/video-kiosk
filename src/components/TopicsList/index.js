// Dependencies
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

// Styles
import './index.css';

const TopicsList = props => (
  <div className="b-topics-list">
    <h2 className="b-topics-list__title">
      Topics
    </h2>
    <ul className="b-topics-list__list">
      {props.categories.map(category => (
        <li
          className="b-topics-list__item"
          key={category.slug}
        >
          <Link
            to={`/category/${category.slug}`}
            className="b-topics-list__link"
          >
            {category.title}
          </Link>
        </li>
      ))}
    </ul>
  </div>
);

TopicsList.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default TopicsList;
