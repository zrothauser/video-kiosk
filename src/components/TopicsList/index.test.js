import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TopicsList from './index';

describe('components', () => {
  describe('MainMenu', () => {
    const mockProps = {
      selectedCategorySlug: 'clyfford-still-basics',
      categories: [
        {
          slug: 'clyfford-still-basics',
          title: 'Clyfford Still Basics',
        },
        {
          slug: 'go-deeper',
          title: 'Go Deeper',
        },
        {
          slug: 'from-the-archives',
          title: 'From the Archives',
        },
      ],
    };

    it('renders without crashing', () => {
      shallow(<TopicsList {...mockProps} />);
    });

    it('renders correctly', () => {
      const TopicsListComponent = renderer.create(
        <Router>
          <TopicsList {...mockProps} />
        </Router>,
      );

      expect(TopicsListComponent).toMatchSnapshot();
    });
  });
});
