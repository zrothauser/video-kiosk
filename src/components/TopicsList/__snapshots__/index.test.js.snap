// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components MainMenu renders correctly 1`] = `
<div
  className="b-topics-list"
>
  <h2
    className="b-topics-list__title"
  >
    Topics
  </h2>
  <ul
    className="b-topics-list__list"
  >
    <li
      className="b-topics-list__item"
    >
      <a
        className="b-topics-list__link b-topics-list__link--selected"
        href="//clyfford-still-basics/"
        onClick={[Function]}
      >
        Clyfford Still Basics
      </a>
    </li>
    <li
      className="b-topics-list__item"
    >
      <a
        className="b-topics-list__link"
        href="//go-deeper/"
        onClick={[Function]}
      >
        Go Deeper
      </a>
    </li>
    <li
      className="b-topics-list__item"
    >
      <a
        className="b-topics-list__link"
        href="//from-the-archives/"
        onClick={[Function]}
      >
        From the Archives
      </a>
    </li>
  </ul>
</div>
`;
